text/0000-javascript-module-api.md

```diff
@@ -0,0 +1,1151 @@
+| `Ember.Checkbox`                      | `import Checkbox from "@ember/component/checkbox"`                         |
+| `Ember.Component`                     | `import Component from "@ember/component"`                                 |
+| `Ember.ContainerDebugAdapter`         | `import ContainerDebugAdapter from "@ember/debug/container-debug-adapter"` |
+| `Ember.Controller`                    | `import Controller from "@ember/controller"`                               |
+| `Ember.DataAdapter`                   | `import DataAdapter from "@ember/debug/data-adapter"`                      |
+| `Ember.DefaultResolver`               | `import GlobalsResolver from "@ember/application/globals-resolver"`        |
+| `Ember.Enumerable`                    | `import Enumerable from "@ember/enumerable"`                               |
+| `Ember.Evented`                       | `import Evented from "@ember/object/evented"`                              |
+| `Ember.HashLocation`                  | `import HashLocation from "@ember/routing/hash-location"`                  |
+| `Ember.Helper`                        | `import Helper from "@ember/component/helper"`                             |
+| `Ember.Helper.helper`                 | `import { helper } from "@ember/component/helper"`                         |
+| `Ember.HistoryLocation`               | `import HistoryLocation from "@ember/routing/history-location"`            |
+| `Ember.Location`                      | `import Location from "@ember/routing/location"`                           |
+| `Ember.Map`                           | `import EmberMap from "@ember/map"`                                        |
+| `Ember.MapWithDefault`                | `import MapWithDefault from "@ember/map/with-default"`                     |
+| `Ember.Mixin`                         | `import Mixin from "@ember/object/mixin"`                                  |
+| `Ember.MutableArray`                  | `import MutableArray from "@ember/array/mutable"`                          |
+| `Ember.NoneLocation`                  | `import NoneLocation from "@ember/routing/none-location"`                  |
+| `Ember.Object`                        | `import EmberObject from "@ember/object"`                                  |
+| `Ember.RSVP`                          | `import RSVP from "rsvp"`                                                  |
+| `Ember.Resolver`                      | `import Resolver from "@ember/application/resolver"`                       |
+| `Ember.Route`                         | `import Route from "@ember/routing/route"`                                 |
+| `Ember.Router`                        | `import Router from "@ember/routing/router"`                               |
+| `Ember.Service`                       | `import Service from "@ember/service"`                                     |
+| `Ember.String.camelize`               | `import { camelize } from "@ember/string"`                                 |
```

Comment:
In the future when we are doing tree-shaking, is this API going to force us to use all of `@ember/string`? From my understanding, the `ember/string/index.js` file is going to look like this in order to support this API:

```js
export { default as camelize } from './camelize';
export { default as capitalize } from './capitalize';
export { default as classify } from './classify';
// ...
```

Which means the entire namespace of `@ember/string` has to be imported so you can use any one piece. I would think to better support slimmer builds, we would only support imports like this:

```js
import camelize from '@ember/string/camelize';
```

Am I mistaken?